### ISSUE TITLE
`template-indent`: Make it also operate on JSX attribute values

### ISSUE THREAD

**Issue by abrahamguo** [sindresorhus/eslint-plugin-unicorn#2886]
### Description

Regular JavaScript string literals (i.e. single- or double-quote strings) cannot contain literal newlines.

However, in JSX attribute values, they CAN contain literal newlines.

It would be nice if the `template-indent` rule could operate on those, as they are functionally the same as template literals.

### Examples

```js
// ❌
<div title='
     Hello
world
'>

// ✅
<div title='
  Hello
  World
'>
```

**Comment by fregante** [comment]
Isn't whitespace there important? Particularly line breaks. What does your second example look like?

**Comment by aguo-ntech** [comment]
Whitespace is important in the `title` attribute specifically, but that was just one example.
The `class` (`className` in React) attribute is multi-whitespace-insensitive.

**Comment by mmkal** [comment]
> Isn't whitespace there important? Particularly line breaks. What does your second example look like in the browser-rendered tooltip?

The rule allows defining [whatever you want as a selector](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/template-indent.md#options), so if we did this we could just leave it up to the user to configure which attributes they want to format this way. They should obviously not enable it for any attributes where whitespace is important in their project. className seems like quite a good candidate as a default.

**Comment by sindresorhus** [comment]
Thanks for the suggestion, but I don't think this belongs in `template-indent`.

The rule is intentionally limited to template literals and explicitly whitelisted template-like cases where whitespace is known to be safe to normalize. JSX attribute values are regular runtime strings, and whether whitespace is significant depends on the specific attribute and component. `className` is usually whitespace-insensitive for DOM elements, but it is still just an arbitrary prop on custom components, and other attributes like `title` are whitespace-sensitive.

Since this rule is recommended and autofixable, expanding it to JSX attributes would make it too easy to silently change runtime string values. I think the simpler and safer boundary is to keep `template-indent` focused on templates, not JSX prop formatting.
